crates/rust-analyzer/src/global_state.rs

```diff
@@ -186,11 +186,9 @@ impl GlobalState {
             }
 
             for file in changed_files {
-                if file.is_created_or_deleted() {
-                    if let Some(path) = vfs.file_path(file.file_id).as_path() {
-                        fs_changes.push((path.to_path_buf(), file.change_kind));
-                        has_fs_changes = true;
```

Comment:
I think this is not correct, or rather, doesn’t match the way this code indents do work.

fs_changes refers to changes to the file system itself, that is, addition or deletion of files. Changes to just file contents are not fs_changes.

motivation example here is that we should run cargo metadata when a new file is added to examples, but we shouldn’t re-run on every typing in this new file

Reply:
Reloading when editing `main.rs` doesn't happen in practice with this PR because of [this check](https://github.com/rust-analyzer/rust-analyzer/blob/master/crates/rust-analyzer/src/reload.rs#L85-L87) in `maybe_refresh`. And without this PR, a second edit to `Cargo.toml` is ignored, because the file is already in the VFS and the change doesn't get to `fs_changes`.

If the workspace reload is triggered by `maybe_refresh`, we need to get the `Cargo.toml` edit in there.